content/using/running/minio.md

```diff
@@ -0,0 +1,176 @@
+
+You can read more about S3 in [Configuring S3 Storage](/using/os/s3).
+
+Cloud providers offer off-the-shelf S3 solutions that you can get started with almost immediately. However, if you do not wish to trust them with your uploaded files, you can self-host an S3 solution.
+
+This process requires a working knowledge of the Linux command line and web technologies such as DNS and TLS.
+
+The self-hosting process is almost exactly the same whether you are hosting on your own hardware, or if you are renting a VPS from a cloud provider like DigitalOcean or AWS. Either way, all that you need is a machine running Linux and a domain - for the sake of this guide, `example.com` will be used in place of your custom domain. You should substitute your own domain everywhere the example domain is used in this guide; for example:
+
+- `example.com` should become `yourdomain.tld`,
+- `s3.example.com` should become `s3.yourdomain.tld`,
+
+and so on.
+
+Ideally, MinIO would be installed on the same machine that your Urbit planet is hosted on, and run alongside at no extra cost - although it can just as easily be hosted on a separate machine.
+
+## Steps
+
+### (optional) Install Docker
+
+Hosting MinIO via Docker is the simplest option. To install Docker server, follow the relevant guide [here](https://docs.docker.com/engine/install/#server).
+
+If you don't want to install Docker, MinIO does offer independent binaries. The process for running these should be similar, but the next step assumes you are using Docker.
+
+### Install MinIO
+
+Once Docker is installed, we can install and run MinIO by following the steps [here](https://docs.min.io/docs/minio-docker-quickstart-guide.html).
+
+You should only need to run a single command, along the lines of:
+
+```
+docker run -d -p 9000:9000 \
+  --name minio-urbit \
+  -v /mnt/data:/data \
+  -e "MINIO_ACCESS_KEY=<access_key>" \
+  -e "MINIO_SECRET_KEY=<secret_key>" \
+  -e "MINIO_DOMAIN=s3.example.com" \
+  minio/minio server /data
+```
+
+`/mnt/data` is the path where your uploaded data will be stored on your host machine - you can change this as necessary.
+
+Be sure to add the `MINIO_DOMAIN` environment variable; this tells MinIO to accept virtual host style URLs (`BUCKET.s3.example.com` rather than `s3.example.com/BUCKET`), which are required for compatibility with Urbit.
+
+Your access key and secret key can be anything of your choosing — make sure they're secure! ~ribben-donnyl recommended a handy tool called [minio-keygen](https://github.com/iwittkau/minio-keygen) that will generate secure keys for you. You could also use your `@p` and `+code` for an easy to remember solution.
+
+### Create DNS records
+
+Now, you'll need to point your own domain at your MinIO installation. Via your domain's DNS settings (usually configured on the registrar you bought your domain through), create two `A` records:
+
+- `s3`.example.com, and
+- `*.s3`.example.com
```

Comment:
Are the backtics supposed to appear in the URL? This should be written for an audience that has never dealt with DNS before and its look strange, so you should clarify this. (I actually don't know the answer to this question - I've only set up DNS once before and don't remember if there were backtics or not)

Reply:
I should clarify that I spaced out and was thinking that the backtics would appear when the page is rendered, rather than just show as code blocks. Either way, I would find it mildly confusing for part of the line to be in a code block and part of it not to be, so I think it should probably just encompass the whole line like `s3.example.com`